src/dotnet-install.ps1

```diff
@@ -638,6 +686,12 @@ else {
     throw "Invalid value for `$Runtime"
 }
 
+if ($SpecificVersion -ne $EffectiveVersion)
```

Comment:
We should move this check before `if(DryRun) {` above. Otherwise, dryrun would yield a different result than an actual run

Reply:
I thought about that but there's this line inside the dry run stuff:

```
$RepeatableCommand = ".\$ScriptName -Version `"$SpecificVersion`" -InstallDir `"$InstallRoot`" -Architecture `"$CLIArchitecture`""
```
So, if I change specific version before this line, this command is no longer valid or repeatable.  How about just logging effective version in the dry run?

Reply:
Why would it be no longer valid?

Reply:
That's because that version isn't the specific version they're trying to download or install, it simply allows the version inside the container to differ from outside (e.g. /5.0.0-rc2.12345.67/ contaning just 5.0.0 versions)

Reply:
I made Dry Run print out the alternate version if it's not the same; proactively resolving this but we can discuss further if needed.